src/Underworld.ts

```diff
@@ -3442,8 +3443,11 @@ ${CardUI.cardListToImages(player.stats.longestSpell)}
       }
     }
   }
-  // TODO - This should factor in fortify, debilitate, bloat explosions, etc.
-  // Any reason we dont fully simulate the enemy turn and store relevant results?
+  // For AI Refactor: https://github.com/jdoleary/Spellmasons/issues/388
+  // TODO - This doesn't factor in fortify, debilitate, bloat explosions, etc.
+  // This also doesn't play well with units that have different actions, such as the Goru
+  // Is there a way for us to better predict the enemy turn, in a way that considers
+  // the game state and always stays in sync with the actual outcome of combat?
```

Comment:
I honestly don't know. I imagine there's a wide range of unpredictable effects, including false negatives where a unit says they won't attack and then they do. But this has been an issue for a long time and only affects certain edge case scenarios where combinations of modifiers and onDeath/onDamage effects are involved. Right now it's not that bad, but it will become increasingly unstable as the game grows and more effects are added.

It's good enough to go for this update / PR I think

Reply:
Can you make a separate issue to track this todo please?

Reply:
Sure! I added this comment and convo to
https://github.com/jdoleary/Spellmasons/issues/409

It's also linked in #388